src/main/java/io/kestra/storage/s3/S3Storage.java

```diff
@@ -69,6 +78,44 @@ public InputStream get(String tenantId, URI uri) throws IOException {
         }
     }
 
+    private boolean exists(String path) throws IOException {
+        try {
+            get(path);
+            return true;
+        } catch (FileNotFoundException exception) {
+            return false;
+        }
+    }
+
+    @Override
+    public List<FileAttributes> list(String tenantId, URI uri) throws IOException {
+        String path = getPath(tenantId, uri);
+        String prefix = (path.endsWith("/")) ? path : path + "/";
+        if (!exists(prefix)) {
+            throw new FileNotFoundException(uri + " (Not Found)");
+        }
```

Comment:
Is it needed? Doesn't the `listObjectsV2` method throw an exception (or return an empty list) so we can avoid calling twice the API.

Reply:
it will return an empty list because it does not check for a directory but for object that start with a prefix... I could remove this but the other storage throws so the test check that it throws...

Reply:
I throw if the return list is empty if that's ok ?